### ISSUE TITLE
On the problem of D_single loss function not converging

### ISSUE THREAD

**Issue by zhatiai** [xiaofeng94/RefineDNet-for-dehazing#6]
Hello，thanks for your work and l'm sorry to bother you.When I used your code to train on the ITS training set, after more than 20 epochs of training, the loss function G_single has been in a fluctuating state and has no convergence trend. And the loss function curve displayed on visdom is as follows:
<img width="375" alt="image" src="https://user-images.githubusercontent.com/86452931/159937423-8b824d1f-f1cc-4852-b897-2249f9d49047.png">
<img width="1014" alt="image" src="https://user-images.githubusercontent.com/86452931/160036386-9d1d1e02-bd71-482a-9964-acfa8e333c60.png">

Below is my command when executing training：
`python train.py --dataroot ./datasets/ITS --dataset_mode unpaired --model refined_DCP --name refined_DCP_indoor --display_server 0.0.0.0 --display_port 6006 --niter 30 --niter_decay 60 --lr_decay_iters 5 --preprocess scale_width_and_crop --load_size 372 --crop_size 256 --num_threads 8 --save_epoch_freq 1 --norm batch --batch_size 16 --lr_policy cosine 
`

At the same time, the training time of more than 20 epochs is as long as 5 hours. Check the graphics card status It can be determined that the training is performed on the graphics card, and the graphics card configuration is as follows:GPU NVIDIA GeForce RTX 3090 数量： 1 显存： 24 GB。So, I don't know whether it is because of my graphics card that the training is slow or your code itself has a long training cycle.

**Comment by xiaofeng94** [comment]
Hi, 

Sorry for the late response. It seems no notification emails come to me. D_single and G_single are loss for GAN and will not converge by design.

**Comment by CharlesLayn** [comment]
Hello, thanks for your excellent work!  I wonder how you can tell which model can achieve the best dehazing effect? Because it seems that the results cannot be observed from the loss printed during the training process, my current practice is to use each model to dehaze the test set, and then use MATLAB to measure the dehazed image, I don't know if this is correct, so I want to ask how you judge which epoch is the best model ?
